src/System.Net.Requests/src/System/Net/FtpWebResponse.cs

```diff
@@ -16,7 +16,7 @@ public class FtpWebResponse : WebResponse, IDisposable
         private Uri _responseUri;
         private FtpStatusCode _statusCode;
         private string _statusLine;
-        private WebHeaderCollection _ftpRequestHeaders;
+        private volatile WebHeaderCollection _ftpRequestHeaders;
```

Comment:
I'd prefer to see volatile not added here, and instead just change the Headers property from:
```C#
        public override WebHeaderCollection Headers
        {
            get
            {
                if (_ftpRequestHeaders == null)
                {
                    lock (this)
                    {
                        if (_ftpRequestHeaders == null)
                        {
                            _ftpRequestHeaders = new WebHeaderCollection();
                        }
                    }
                }
                return _ftpRequestHeaders;
            }
        }
```
to instead be:
```C#
        public override WebHeaderCollection Headers =>
            LazyInitializer.EnsureInitialized(ref _ftpRequestHeaders);
```